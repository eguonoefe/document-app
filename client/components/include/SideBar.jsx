import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
/**
* Sidebar component that displays a Sidebar.
* @param {Object} props { match, user }
* @returns {String} The HTML markup for the Sidebar component
*/
const SideBar = ({ match, user }) => (
  <ul id="slide-out" className="side-nav fixed ">
    <li><div className="user-view">
      <div className="side-logo center">
        <Link to="/" className="brand-logo">
          <img
            className="brand-img"
            src="../../img/logo2.jpg" alt="sidebar-logo" /> ms</Link>
      </div>
    </div></li>
    <ul className="collapsible" data-collapsible="accordion">
      <li>
        <div className="collapsible-header" id="document-header">
          <i className="material-icons">note</i>Document</div>
        <div className="collapsible-body" id="document">
          <ul>
            <li><Link to={`${match.url}/document`} id="create-document">
              <i className="material-icons">note_add</i>
              Create Document</Link>
            </li>
            <li><Link to={`${match.url}/all-documents`}>
              <i className="material-icons">perm_media</i>
              All Document</Link></li>
          </ul>
        </div>
      </li>
      <li>
        <div className="collapsible-header" id="user-header">
          <i className="material-icons">people</i>Users</div>
        <div className="collapsible-body" id="user">
          <ul>
            <li><Link to={`${match.url}/profile`} id="profile">
              <i className="material-icons">perm_identity</i>
              Profile</Link></li>
            {user.roleId === 1 && <li>
              <Link to={`${match.url}/all-users`} id="userList">
                <i className="material-icons">people_outline</i>
                All Users</Link></li>}
          </ul>
        </div>
      </li>
      {user.roleId === 1 && <li>
        <div className="collapsible-header" id="role-header">
          <i className="material-icons">face</i>Roles</div>
        <div className="collapsible-body">
          <ul>
            {user.roleId === 1 && <li><Link
              to={`${match.url}/role`}
              id="create-role">
              <i className="material-icons">create</i>
              Create Role</Link>
            </li>}
            {user.roleId === 1 && <li><Link
              to={`${match.url}/all-roles`}
              id="roleList">
              <i className="material-icons">wc</i>
              All Roles</Link></li>}
          </ul>
        </div>
      </li>}
    </ul >
  </ul >
);

SideBar.propTypes = {
  match: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired
};

export default SideBar;

