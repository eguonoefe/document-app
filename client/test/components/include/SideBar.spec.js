/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import SideBar from '../../../components/include/SideBar';

configure({ adapter: new Adapter() });

jest.dontMock('../../../components/include/SideBar');

describe('SideBar Component', () => {
  test('should match the sidebar snapshot', () => {
    const match = {
      url: '/'
    };
    const user = {
      roleId: 1
    };
    const component = shallow(<SideBar match={match} user={user} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
