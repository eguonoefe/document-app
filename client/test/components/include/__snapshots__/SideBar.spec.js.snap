// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component should match the sidebar snapshot 1`] = `
<ul
  className="side-nav fixed "
  id="slide-out"
>
  <li>
    <div
      className="user-view"
    >
      <div
        className="side-logo center"
      >
        <Link
          className="brand-logo"
          to="/"
        >
          <img
            alt="sidebar-logo"
            className="brand-img"
            src="../../img/logo2.jpg"
          />
           ms
        </Link>
      </div>
    </div>
  </li>
  <ul
    className="collapsible"
    data-collapsible="accordion"
  >
    <li>
      <div
        className="collapsible-header"
        id="document-header"
      >
        <i
          className="material-icons"
        >
          note
        </i>
        Document
      </div>
      <div
        className="collapsible-body"
        id="document"
      >
        <ul>
          <li>
            <Link
              id="create-document"
              to="//document"
            >
              <i
                className="material-icons"
              >
                note_add
              </i>
              Create Document
            </Link>
          </li>
          <li>
            <Link
              to="//all-documents"
            >
              <i
                className="material-icons"
              >
                perm_media
              </i>
              All Document
            </Link>
          </li>
        </ul>
      </div>
    </li>
    <li>
      <div
        className="collapsible-header"
        id="user-header"
      >
        <i
          className="material-icons"
        >
          people
        </i>
        Users
      </div>
      <div
        className="collapsible-body"
        id="user"
      >
        <ul>
          <li>
            <Link
              id="profile"
              to="//profile"
            >
              <i
                className="material-icons"
              >
                perm_identity
              </i>
              Profile
            </Link>
          </li>
          <li>
            <Link
              id="userList"
              to="//all-users"
            >
              <i
                className="material-icons"
              >
                people_outline
              </i>
              All Users
            </Link>
          </li>
        </ul>
      </div>
    </li>
    <li>
      <div
        className="collapsible-header"
        id="role-header"
      >
        <i
          className="material-icons"
        >
          face
        </i>
        Roles
      </div>
      <div
        className="collapsible-body"
      >
        <ul>
          <li>
            <Link
              id="create-role"
              to="//role"
            >
              <i
                className="material-icons"
              >
                create
              </i>
              Create Role
            </Link>
          </li>
          <li>
            <Link
              id="roleList"
              to="//all-roles"
            >
              <i
                className="material-icons"
              >
                wc
              </i>
              All Roles
            </Link>
          </li>
        </ul>
      </div>
    </li>
  </ul>
</ul>
`;
